commit c0ab1469dfffee11468c91077f69cbe8e0adf087 (refs/changes/01/3501/3)
Author: Tim Kordas <tim.kordas@joyent.com>
Date:   2018-04-20T22:47:58+00:00 (1 year, 6 months ago)
    
    OS-6561: reboot (or unmount) inside zone while tailing file from GZ leaks fem handler(s)

diff --git a/usr/src/uts/common/fs/fem.c b/usr/src/uts/common/fs/fem.c
index 5f524def30..2a80144486 100644
--- a/usr/src/uts/common/fs/fem.c
+++ b/usr/src/uts/common/fs/fem.c
@@ -4035,8 +4035,11 @@ fsem_install(
 	int	error;
 	struct fem_node	nnode;
 
-	/* If this vfs hasn't been properly initialized, fail the install */
-	if (vfsp->vfs_implp == NULL)
+	/*
+	 * If this vfs hasn't been properly initialized or has been unmounted,
+	 * fail the install.
+	 */
+	if (vfsp->vfs_implp == NULL || (vfsp->vfs_flag & VFS_UNMOUNTED))
 		return (EINVAL);
 
 	nnode.fn_available = arg;
